scenarioo-server/src/main/java/org/scenarioo/rest/search/SearchResource.java

```diff
@@ -0,0 +1,56 @@
+/* scenarioo-server
+ * Copyright (C) 2014, scenarioo.org Development Team
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.scenarioo.rest.search;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import org.scenarioo.business.builds.ScenarioDocuBuildsManager;
+import org.scenarioo.dao.search.FullTextSearch;
+import org.scenarioo.model.docu.entities.generic.ObjectReference;
+import org.scenarioo.model.docu.entities.generic.ObjectTreeNode;
+import org.scenarioo.rest.base.BuildIdentifier;
+
+@Path("/rest/branch/{branchName}/build/{buildName}/")
+public class SearchResource {
+
+	@GET
+	@Produces("application/json")
+	@Path("/search/{q}")
+	public ObjectTreeNode<ObjectReference> search(@PathParam("branchName") final String branchName,
+												  @PathParam("buildName") final String buildName, @PathParam("q") final String q) {
+
+		BuildIdentifier buildIdentifier = ScenarioDocuBuildsManager.INSTANCE.resolveBranchAndBuildAliases(branchName,
+				buildName);
+
+		FullTextSearch search = new FullTextSearch();
+		return search.search(q, buildIdentifier).buildObjectTree();
+	}
+
+	@GET
+	@Path("/searchEngine")
+	public Response isEngineRunning() {
```

Comment:
This special call should not be bound to a branch and build as parameters in the URL (see path on the whole class) that are then not mapped at all,  since it only generally checks the availability of a search engine in general.

Also somehow I think it should simply return false and not an error code if the engine is not there. Status.NOT_FOUND is for resources that are not there, which is something differently. Or if you really think the status code is the better solution, then we should maybe rename it to "searchEngineStatus" instead of "isEngineRunning", which makes it more clear that no boolean is returned.